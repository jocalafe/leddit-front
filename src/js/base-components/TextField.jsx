import React from 'react';
import TextField from 'material-ui/TextField';
import PropTypes from 'prop-types';

const LedditTextField = ({ input, hint, type, ...rest }) => (
  <div>
    <TextField
      {...input}
      hintText={hint}
      floatingLabelText={hint}
      type={type}
      {...rest}/>
  </div>
);

LedditTextField.propTypes = {
  id: PropTypes.string,
  hint: PropTypes.string,
  disabled: PropTypes.bool,
};

export default LedditTextField;
