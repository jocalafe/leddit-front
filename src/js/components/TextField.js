import React from 'react';
import TextField from 'material-ui/TextField';
import PropTypes from 'prop-types';

const MyTextField = ({ id, hint = '', disabled = false }) => (
  <div>
    <TextField id={id} disabled={disabled} hint={hint} />
  </div>
);

MyTextField.propTypes = {
  id: PropTypes.string,
  hint: PropTypes.string,
  disabled: PropTypes.bool,
};

export default MyTextField;
