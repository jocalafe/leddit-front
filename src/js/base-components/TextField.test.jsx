import React from 'react';
import renderer from 'react-test-renderer';

import TextField from './TextField';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

jest.mock('react-dom');

describe('TextField', () => {
  it('renders', () => {
    const tree = renderer.create(
      <MuiThemeProvider>
        <TextField id='test1' />
      </MuiThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
