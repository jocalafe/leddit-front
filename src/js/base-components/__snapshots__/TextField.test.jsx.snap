// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField renders 1`] = `
<div>
  <div
    className={undefined}
    style={
      Object {
        "backgroundColor": "transparent",
        "cursor": "auto",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 16,
        "height": 48,
        "lineHeight": "24px",
        "muiPrepared": true,
        "position": "relative",
        "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": 256,
      }
    }
  >
    <input
      disabled={false}
      id="test1"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "appearance": "textfield",
          "backgroundColor": "rgba(0,0,0,0)",
          "border": "none",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "inherit",
          "font": "inherit",
          "height": "100%",
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "width": "100%",
        }
      }
      type="text"
    />
    <div>
      <hr
        aria-hidden="true"
        style={
          Object {
            "borderBottom": "solid 1px",
            "borderColor": "#e0e0e0",
            "borderLeft": "none",
            "borderRight": "none",
            "borderTop": "none",
            "bottom": 8,
            "boxSizing": "content-box",
            "margin": 0,
            "muiPrepared": true,
            "position": "absolute",
            "width": "100%",
          }
        }
      />
      <hr
        aria-hidden="true"
        style={
          Object {
            "borderBottom": "solid 2px",
            "borderColor": "#00bcd4",
            "borderLeft": "none",
            "borderRight": "none",
            "borderTop": "none",
            "bottom": 8,
            "boxSizing": "content-box",
            "margin": 0,
            "muiPrepared": true,
            "position": "absolute",
            "transform": "scaleX(0)",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
</div>
`;
